src/OpenApi/src/Comparers/OpenApiAnyComparer.cs

```diff
@@ -30,7 +30,7 @@ public bool Equals(IOpenApiAny? x, IOpenApiAny? y)
             {
                 OpenApiNull _ => y is OpenApiNull,
                 OpenApiArray arrayX => y is OpenApiArray arrayY && arrayX.SequenceEqual(arrayY, Instance),
-                OpenApiObject objectX => y is OpenApiObject objectY && objectX.Keys.Count == objectY.Keys.Count && objectX.Keys.All(key => objectY.ContainsKey(key) && Equals(objectX[key], objectY[key])),
+                OpenApiObject objectX => y is OpenApiObject objectY && objectX.Keys.Count == objectY.Keys.Count && objectX.Keys.All(key => objectY.TryGetValue(key, out var yValue) && Equals(objectX[key], yValue)),
```

Comment:
Strange, we have CA1854 enabled. Wonder if the analyzer doesn't find it if it's not an if statement.